zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -248,6 +249,138 @@ def _get_services(cls):
+        result_resource_ids = {resource['id'] for resource in
+                               resources_from_action}
+
+        # assert that juju action returned expected resources
+        self.assertEqual(result_resource_ids, expected_resource_ids)
+
+    def test_get_status_routers(self):
+        """Test that get-status-routers reports correct neutron routers."""
+        # fetch neutron routers using neutron client
+        ngw_unit = zaza.model.get_units(self.application_name,
+                                        model_name=self.model_name)[0]
+        routers_from_client = self.neutron_client.list_routers().get(
+            'routers', [])
+
+        if not routers_from_client:
+            self.fail('At least one router must be configured for this test '
+                      'to pass.')
+
+        # fetch neutron routers using juju-action
+        result = zaza.model.run_action(ngw_unit.entity_id,
+                                       'get-status-routers',
+                                       model_name=self.model_name,
+                                       action_params={"format": "json"})
+
+        # assert that data from neutron client match data from juju action
+        self._assert_result_match(result, routers_from_client, 'router-list')
+
+    def test_get_status_dhcp(self):
+        """Test that get-status-dhcp reports correct DHCP networks."""
+        # fetch DHCP networks using neutron client
+        ngw_unit = zaza.model.get_units(self.application_name,
+                                        model_name=self.model_name)[0]
+        networks_from_client = self.neutron_client.list_networks().get(
+            'networks', [])
+
+        if not networks_from_client:
+            self.fail('At least one network must be configured for this test '
+                      'to pass.')
+
+        # fetch DHCP networks using juju-action
+        result = zaza.model.run_action(ngw_unit.entity_id,
+                                       'get-status-dhcp',
+                                       model_name=self.model_name,
+                                       action_params={"format": "json"})
+
+        # assert that data from neutron client match data from juju action
+        self._assert_result_match(result, networks_from_client,
+                                  'dhcp-networks')
+
+    def test_get_status_load_balancers(self):
+        """Test that get-status-lb reports correct loadbalancers."""
+        if self.SKIP_LBAAS_TESTS:
+            self.skipTest('LBaasV2 is not supported in this version.')
+
+        # create LBaasV2 for the purpose of this test
+        lbaas_name = 'test_lbaas'
+
+        subnet_list = self.neutron_client.list_subnets(
+            name='private_subnet').get('subnets', [])
+
+        if not subnet_list:
+            raise RuntimeError('Expected subnet "private_subnet" is not '
+                               'configured.')
```

Comment:
As above, skip or fail?  The reason for asking, is whether this can be a generic test or is very specific to this environment.